### PR TITLE
bugfix : options can't be passed to SeparatorToSeparator 

### PR DESCRIPTION
**Pull request by Tauop**
Hi,

This PR fixes #6666 .

Patrick

### CODE REVIEW

File: library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    public function createService(ServiceLocatorInterface $plugins)
+    {
+        return new SeparatorToSeparator(
+            isset($this->options['search_separator']) ? $this->options['search_separator'] : ' ',
+            isset($this->options['replacement_separator']) ? $this->options['replacement_separator'] : '-'
+        );
+    }
+
+}
```
Comment (malukenho) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: EOL EOF

File: library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+class SeparatorToSeparatorFactory implements FactoryInterface
+{
+    protected $options;
+
+    /**
+     * Constructor.
+     *
+     * @param array $options
+     */
+    public function __construct($options = array()) {
```
Comment (malukenho) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: Maybe `public function __construct(array $options = array())`

File: library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        $this->options = $options;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return Forward
```
Comment (malukenho) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: `@return SeparatorToSeparator`, nops?

File: library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php (hunk 2 above)
Comment (malukenho) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: Also, put `{` in a new line. Follow the PSR.

Comment (Tauop) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: corrected. thank you.

File: library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php (hunk 3 above)
Comment (Tauop) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: corrected. thank you.
Sorry for the CC mistake :-/

File: library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+     * @var array
+     */
+    protected $options;
+
+    /**
+     * Constructor.
+     *
+     * @param array $options
+     */
+    public function __construct(array $options = array())
```
Comment (Ocramius) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: What happens if I try to get multiple instances of a `wordseparatortoseparator`, each with different options?

Comment (Ocramius) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: Hint: factory must implement the [`MutableCreationOptionsInterface`](https://github.com/zendframework/zf2/blob/9d86b442bddd728612280bddcfe116d368367cea/library/Zend/ServiceManager/MutableCreationOptionsInterface.php)

Comment (Tauop) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: Thank you very much for helping me to make this PR :-)
This was corrected.

Comment (Ocramius) [inline library/Zend/Filter/Word/Service/SeparatorToSeparatorFactory.php]: This constructor is also not needed anymore ;-)

File: tests/ZendTest/Filter/FilterPluginManagerTest.php [hunk 5]
```diff
@@ -39,4 +39,38 @@ public function testLoadingInvalidFilterRaisesException()
         $this->setExpectedException('Zend\Filter\Exception\RuntimeException');
         $this->filters->get('test');
     }
+
+    public function testFilterSuccessfullyConstructed()
```
Comment (Ocramius) [inline tests/ZendTest/Filter/FilterPluginManagerTest.php]: Consider adding `@group` annotations here

File: tests/ZendTest/Filter/FilterPluginManagerTest.php [hunk 6]
```diff
@@ -39,4 +39,38 @@ public function testLoadingInvalidFilterRaisesException()
...
+            'search_separator'      => $search_separator,
+            'replacement_separator' => $replacement_separator,
+        );
+        $filter = $this->filters->get('wordseparatortoseparator', $options);
+        $this->assertInstanceOf('Zend\Filter\Word\SeparatorToSeparator', $filter);
+        $this->assertEquals(';', $filter->getSearchSeparator());
+        $this->assertEquals('|', $filter->getReplacementSeparator());
+    }
+
+    public function testFiltersConstructedAreDifferent()
```
Comment (Ocramius) [inline tests/ZendTest/Filter/FilterPluginManagerTest.php]: Same as above

**Comment by Ocramius** [comment]
@Tauop merged, thanks!

`master`: zendframework/zf2@9ee2075758265876b294a14287c135c63ff241f7
`develop`: zendframework/zf2@2515b0dfde663fea89b30834c29f0eb2d4db3867

**Pull request closed without merging**